src/main/java/forestry/farming/logic/FarmLogicArboreal.java

```diff
@@ -162,38 +164,41 @@ private Collection<ICrop> getHarvestBlocks(Vect position) {
         return crops;
     }
 
-    private ArrayList<Vect> processHarvestBlock(IFarmable germling, Stack<ICrop> crops, Set<Vect> seen, Vect start,
+    private ArrayList<Vect> processHarvestBlock(IFarmable germling, Deque<ICrop> crops, Set<Vect> seen, Vect start,
             Vect position) {
 
         World world = getWorld();
 
         ArrayList<Vect> candidates = new ArrayList<>();
 
         // Get additional candidates to return
+        MutableVect mutable = new MutableVect();
         for (int x = -1; x < 2; x++) {
             for (int y = -1; y < 2; y++) {
                 for (int z = -1; z < 2; z++) {
-                    Vect candidate = position.add(x, y, z);
-                    if (candidate.equals(position)) {
+                    // Vect candidate = position.add(x, y, z);
+                    mutable.set(position).add(x, y, z);
+                    if (mutable.equals(position)) {
                         continue;
                     }
-                    if (Math.abs(candidate.x - start.x) > BRANCH_RANGE) {
+                    if (Math.abs(mutable.x - start.x) > BRANCH_RANGE) {
                         continue;
                     }
-                    if (Math.abs(candidate.z - start.z) > BRANCH_RANGE) {
+                    if (Math.abs(mutable.z - start.z) > BRANCH_RANGE) {
                         continue;
                     }
 
                     // See whether the given position has already been processed
-                    if (seen.contains(candidate)) {
+                    if (seen.contains(mutable)) {
```

Comment:
`seen` is defined as `Set<Vect>` while this check attempts to find `MutableVect`. I know that `contains` receives argument as `Object` and compares hash codes, but to make things clear - `seen` should be `Set<IVect>` to properly work with both types.

Reply:
I originally had Set as an IVect, but I don't want MutableVects to accidently be inserted, because they could change invalidate the requirements of the set.  I ensured the hash and equals function of both will properly identify a vect even if asked about a mutable vect.   I also didn't want the mutable to need to be converted to an immutable (allocation) until the last possible place it could be.

If possible, I'd rather mark Vect & MutableVect as comparable, but I'm not 100% sure how to do that in java.

Reply:
As long as the two classes implement hashCode() identically and equals() allows for comparing with either vect type, `Hash*` containers will allow you to look up one using the other - this is safe and documented in their contract: <https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html#contains-java.lang.Object->. If you implement `Comparable<IVect>` where IVect is an interface implemented by both types, tree collections will work too - if you don't, I'd say you should use a `HashSet` as the field type explicitly with a comment explaining that you need heterogeneous lookup to work (that's how c++ calls this feature, I couldn't find a java name).

Reply:
Other documentation that points to the contracts required is Object's hashCode and equals requirements: <https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode-->

Reply:
I knew about that feature. I just want to
* either clearly mark that mentioned `Set` works with any implementation (`Set<Vect>` -> `Set<IVect>`),
* or to add documentation to `IVect` interface to require implementation of `equals` and `hashCode` compatible to Vect and MutableVect classes. Adding commnet for mentioned Sets would be nice but not necessary.

Reply:
I'd agree with adding a comment on the IVect interface. Sets of mutable objects are prone to weird errors, so I think keeping the Set element type as the immutable vector is better.

Reply:
👍 done